dagql/types.go

```diff
@@ -142,8 +142,6 @@ func (i Int) TypeDefinition(views ...string) *ast.Definition {
 
 func (Int) DecodeInput(val any) (Input, error) {
 	switch x := val.(type) {
-	case nil:
-		return NewInt(0), nil
```

Comment:
x-linking to https://github.com/dagger/dagger/pull/8331, where this behavior was introduced. A better fix for this is just to update `ConvertFromSDKResult` to handle the weird default constructor case.